### DISCUSSION TITLE
Is there a way to find a symmetry pair of the molecule?

### DISCUSSION THREAD

**Discussion by tsn2027** [rdkit/rdkit#6632] [Q&A] [answered]
Hello all,

I wonder if there is an approach to print out a symmetry pair of the molecule? What I have done so far is to print out all atoms and their atom neighbors and then subtract the first and last atom of the list until there is no more atom left; however, I think this approach is insufficient and sometimes it finds incorrect pairs. Thus, I need to find a better way to do this.

What I am looking for is shown below:

```sample_smiles = "CCCCCC" # C6 ```
In this sample, symmetry pairs are ```(0,5),(1,4),(2,3)```

```sample_smiles = "CCCCC" # C5 ```
In this sample, symmetry pairs are ```(0,4),(1,3),(2,)```

```sample_smiles = "CCC(CCC)OO" # C5```
In this sample, symmetry pairs are ```(0,5),(1,4),(3,)(2,)```

However, when it comes to this smiles ```"C1CCCCC1C"```, my function fails.

I am using packages:
```python version: 3.10.12; rdkit version:  2023.03.2```

I'd appreciate the helps! And thanks in advance!

**Comment by iuhgnor** [comment, marked as answer]
`CanonicalRankAtoms` may help. 

By the way, is there a mistake in the smiles of the third case? The smiles has six C atoms.

```python
smi = 'C1CCCCC1C'
mol = Chem.MolFromSmiles(smi)

symm_dict = {}
for i, order in enumerate(Chem.CanonicalRankAtoms(mol, breakTies=False)):
    if order not in symm_dict:
        symm_dict[order] = []
    symm_dict[order].append(i)

list(symm_dict.values())
```

```txt
[[0, 4], [1, 3], [2], [5], [6]]
```

**Reply by tsn2027** [reply]
@iuhgnor Thank you!! I tried to use ```CanonicalRankAtoms``` with ```breakTies=False``` before but could not find any useful info (my function only determines whether it is symmetric or not).

Also, you are right, the 3rd smiles should have 6 C atoms.
